test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -371,6 +379,32 @@ func TestVaultDynamicSecret(t *testing.T) {
 	})
 }
 
+// assertDynamicSecretNewGeneration tests that an update to vdsObjOrig results in a full secret rotation.
+func assertDynamicSecretNewGeneration(t *testing.T, ctx context.Context, client ctrlclient.Client, vdsObjOrig *secretsv1alpha1.VaultDynamicSecret) {
+	t.Helper()
+
+	objKey := ctrlclient.ObjectKeyFromObject(vdsObjOrig)
+	vdsObjLatest := &secretsv1alpha1.VaultDynamicSecret{}
+	if assert.NoError(t, client.Get(ctx, objKey, vdsObjLatest)) {
+		vdsObjLatest.Spec.Destination.Name += "-new"
+		if assert.NoError(t, client.Update(ctx, vdsObjLatest)) {
+			assertDynamicSecretRotation(t, ctx, client, vdsObjLatest)
```

Comment:
I'm just trying to follow this, we issue a `client.Update()` on the vdsObj, changing it's destination secret, which should trigger a reconciliation + creation of the new k8s secret.
But the revocation of the lease as part of `assertDynamicSecretRotation` is actually what triggers the rotation right?

It seems to me that either a lease revocation or the changing of the destination.name should trigger this (and a rollout restart)?
Am I confusing myself?